homeassistant/components/homewizard/coordinator.py

```diff
@@ -42,9 +46,24 @@ async def _async_update_data(self) -> DeviceResponseEntry:
             }
 
         except RequestError as ex:
-            raise UpdateFailed("Device did not respond as expected") from ex
+            raise UpdateFailed(ex) from ex
 
         except DisabledError as ex:
-            raise UpdateFailed("API disabled, API must be enabled in the app") from ex
+            if not self.api_disabled and self.config_entry:
+                self.api_disabled = True
+                self.config_entry.async_start_reauth(self.hass)
+
+            # Exponentially increase update interval
+            if (
+                self.update_interval is not None
+                and self.update_interval < MAX_UPDATE_INTERVAL
+            ):
+                self.update_interval = self.update_interval * 2
```

Comment:
An alternative to doing it like this, is just reloading the config entry. And during setup of the config entry, it will raise ReauthNeeded. HA will automatically do the backoff.

Reply:
I agree it is better to use a build-in process. But if we use this suggestion we loose the 'feature' that when the integration loads when the API is disabled that it just stops trying.

After thinking a bit about this, we only try to prevent issues when users turn off and on the API for a short moment (eg. he was playing in the app and tries the toggle). I think that when the integration is (re)loaded that the user knows about the situation.

Maybe we just want to just retry for a short moment (30 minutes) and then just stop? Something like:

```
except DisabledError:
  if time < 30:
    raise UpdateFailed

  raise ConfigEntryAuthFailed
```

Reply:
why would we treat it differently if the API is disabled on startup vs when it's disabled while HA is running? I think that it's fine to keep trying in both cases.

Reply:
If a user doesn't want to see the error and wants HA to stop trying -> disable/remove config entry. If they want to fix -> enable local API.

Reply:
Fair enough, will work on that asap!